### PR TITLE
phpci:daemon cleanup

### PR DESCRIPTION
**Pull request by Adirelle**
Reworked the DaemonCommand.
* Accepts options for the PID and log file.
* Uses the best process control method available on the installation : POSIX, windows CLI or Unix CLI.
* Checks the daemon actually started or stopped.
* Try to terminate then kill the daemon.
* Uses either the logger or the output interface.

Notes:
* this PR is just a few commits extracted from #864.
* IMO, the log file should be handled directly in the phpci:daemonise command using LoggerConfig.

### CODE REVIEW

**Comment by mavimo** [comment]
Since the last commit is not builded from PHPCI i removed the label, waiting for a new build.

**Comment by mavimo** [comment]
@Adirelle can you do a small change and push again in order to re-run the build? Thx!

**Comment by Adirelle** [comment]
@mavimo ping !

File: PHPCI/ProcessControl/ProcessControl.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+namespace PHPCI\ProcessControl;
+
+/**
+ * A stateless service to check and kill system processes.
+ *
+ * @author Adirelle <adirelle@gmail.com>
+ */
+interface ProcessControl
```
Comment (mavimo) [inline PHPCI/ProcessControl/ProcessControl.php]: I prefer to have ```ProcessControlInterface``` as name, is possible to change it?

**Pull request closed without merging**